samples/cpp/digits.cpp

```diff
@@ -1,375 +1,185 @@
-#include "opencv2/core.hpp"
-#include "opencv2/highgui.hpp"
-#include "opencv2/imgcodecs.hpp"
-#include "opencv2/imgproc.hpp"
-#include "opencv2/ml.hpp"
+//  This example provides a digital recognition based on LeNet-5 and connected component analysis.
+//  It makes it possible for OpenCV beginner to run dnn models in real time using only CPU.
+//  It can read pictures from the camera in real time to make predictions, and display the recognized digits as overlays on top of the original digits.
+//
+//  In order to achieve a better display effect, please write the number on white paper and occupy the entire camera.
+//
+//  You can follow the following guide to train LeNet-5 by yourself using the minist dataset.
+//  https://github.com/intel/caffe/blob/a3d5b022fe026e9092fc7abc7654b1162ab9940d/examples/mnist/readme.md
+//
+//  You can also download and train the model directly.
+//  https://github.com/zihaomu/opencv_lenet_demo/blob/master/src/lenet.caffemodel
+//  https://github.com/zihaomu/opencv_lenet_demo/blob/master/src/lenet.prototxt
+
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/dnn.hpp>
 
-#include <algorithm>
 #include <iostream>
 #include <vector>
 
-using namespace cv;
 using namespace std;
```

Comment:
Please remove `sing namespace std;` - it's bad practice.

Reply:
fixed